### DISCUSSION TITLE
Inverted/negativ colors in GrabberPreview

### DISCUSSION THREAD

**Discussion by sysTim1977** [awawa-dev/HyperHDR#797] [Q&A] [answered]
After upgrade to the latest kernel (Debian Bookworm on RaspberryPi Zero2w, kernel:6.6.20+rpt-rpi-v8) the grabber (MS2131 based) shows inverted/negative colors. I think it is a v4l2-subsystem issue, but perhaps anybody has same problems and can show me a quick solution. Everything else works fine.

PS: Thanks 4 that great ambilight software

**Comment by awawa-dev** [comment]
Hi
Negative or swapped (for example red -> blue)?

**Reply by sysTim1977** [reply]
Thanks for quick answer.
The grabberoutput/preview is negative/inverted
It is no led issue, its a grabber issue.
Before Kernelupdate everything works fine.

**Reply by awawa-dev** [reply]
upload screenshot from live preview

**Comment by sysTim1977** [comment]
![shot232247](https://github.com/awawa-dev/HyperHDR/assets/163929358/1e8f1cc4-b2c3-4721-acc7-00493573794a)

**Comment by awawa-dev** [comment]
I dont think they are inverted but colors are swapped. If it happened suddenly, it may be related to endianness configuration of your OS. We expect little-endian.

**Comment by sysTim1977** [comment]
It happens after kernelupgrade on the ambilightsystem. 
The ambilightsystem is a headless  RaspberrypiZero2w with Debian Bookworm, 64bit and a MS2131 based grabber.
As i said, everything was fine before upgrade. So i dont think it is a HyperHDR problem, it is a v4l2 issue. Perhaps anybody can show me, how to fix thin in the v4l2-subsystem. 
It was a plain RaspiOS-Lite installation (some days before, kernel 6.1(?))) and after the kernelupgrade, i had the colorproblem.

**Reply by sysTim1977** [reply, marked as answer]
It is defintly a Kernel Problem, I Downgrade to 6.1 and everythink is fine again.
